### PR TITLE
Add stecha keymaps for crkbd, boardwalk, xd75 and preonic

### PR DESCRIPTION
**Pull request by SteKarab**
## Description

Add personal keymaps for the following keyboards:

- crkbd
- boardwalk
- xd75
- preonic/rev3

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [ ] Keyboard (addition or update)
- [x] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* none

## Checklist

- [ ] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [ ] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/boardwalk/keymaps/stecha/keymap.c [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include QMK_KEYBOARD_H
+
+// Layer shorthand
+#define _QW 0
+#define _NUM 1
+#define _SYM 2
```
Comment (drashna) [inline keyboards/boardwalk/keymaps/stecha/keymap.c:21]:
```suggestion
enum layers {
    _QW,
    _NUM,
    _SYM,
};
```

File: keyboards/crkbd/keymaps/stecha/keymap.c [hunk 2]
```diff
@@ -0,0 +1,182 @@
+#include QMK_KEYBOARD_H
+
+#ifdef RGBLIGHT_ENABLE
+// Following line allows macro to read current RGB settings
+extern rgblight_config_t rgblight_config;
+#endif
```
Comment (drashna) [inline keyboards/crkbd/keymaps/stecha/keymap.c:6]: This isn't needed, and should be removed. 
```suggestion
```

File: keyboards/crkbd/keymaps/stecha/keymap.c [hunk 3]
```diff
@@ -0,0 +1,182 @@
...
+
+const uint16_t PROGMEM keymaps[][MATRIX_ROWS][MATRIX_COLS] = {
+
+    [_QWERTY] = LAYOUT(KC_ESC, KC_Q, KC_W, KC_E, KC_R, KC_T, KC_Y, KC_U, KC_I, KC_O, KC_P, KC_BSPC, LCTL_T(KC_TAB), KC_A, KC_S, KC_D, KC_F, KC_G, KC_H, KC_J, KC_K, KC_L, KC_SCLN, KC_QUOT, LSFT_T(KC_BSLS), KC_Z, KC_X, KC_C, KC_V, KC_B, KC_N, KC_M, KC_COMM, KC_DOT, KC_SLSH, KC_RSFT, DSKP_LEFT, NUM, KC_SPC, KC_ENT, SYM, DSKP_RIGHT),
+
+    [_NUM] = LAYOUT(KC_ESC, KC_1, KC_2, KC_3, KC_4, KC_5, KC_6, KC_7, KC_8, KC_9, KC_0, KC_BSPC, LCTL_T(KC_TAB), KC_LT, KC_HOME, KC_PGUP, KC_PGDN, KC_END, KC_LEFT, KC_DOWN, KC_UP, KC_RGHT, KC_GT, KC_TAB, LSFT_T(KC_GRV), KC_UNDS, KC_LPRN, KC_DEL, KC_MINS, KC_LBRC, KC_RBRC, KC_EQL, KC_INS, KC_RPRN, KC_PLUS, KC_RSFT, DSKP_LEFT, NUM, KC_SPC, KC_ENT, SYM, DSKP_RIGHT),
+
+    [_SYM] = LAYOUT(KC_ESC, KC_EXLM, KC_AT, KC_HASH, KC_DLR, KC_PERC, KC_CIRC, KC_AMPR, KC_ASTR, KC_LPRN, KC_RPRN, KC_BSPC, KC_F1, KC_F2, KC_F3, KC_F4, KC_F5, KC_F6, KC_F7, KC_F8, KC_F9, KC_F10, KC_F11, KC_F12, LSFT_T(KC_GRV), RGB_MOD, RESET, RGB_HUD, RGB_SAD, RGB_VAD, RGB_VAI, RGB_SAI, RGB_HUI, KC_NO, RGB_TOG, KC_RSFT, DSKP_LEFT, NUM, KC_SPC, KC_ENT, SYM, DSKP_RIGHT)};
+
+int RGB_current_mode;
```
Comment (drashna) [inline keyboards/crkbd/keymaps/stecha/keymap.c:32]:
```suggestion
uint8_t RGB_current_mode;
```

File: keyboards/crkbd/keymaps/stecha/keymap.c [hunk 4]
```diff
@@ -0,0 +1,182 @@
...
+
+void persistent_default_layer_set(uint16_t default_layer) {
+    eeconfig_update_default_layer(default_layer);
+    default_layer_set(default_layer);
+}
+
```
Comment (drashna) [inline keyboards/crkbd/keymaps/stecha/keymap.c:38]: Could you remove this function?  

It's been deprecated in favor of the `set_single_persistent_default_layer` function (which is a core function).

https://docs.qmk.fm/#/ref_functions?id=setting-the-persistent-default-layer

File: keyboards/crkbd/keymaps/stecha/keymap.c [hunk 5]
```diff
@@ -0,0 +1,182 @@
...
+// Setting ADJUST layer RGB back to default
+void update_tri_layer_RGB(uint8_t layer1, uint8_t layer2, uint8_t layer3) {
+    if (IS_LAYER_ON(layer1) && IS_LAYER_ON(layer2)) {
+        layer_on(layer3);
+    } else {
+        layer_off(layer3);
+    }
+}
```
Comment (drashna) [inline keyboards/crkbd/keymaps/stecha/keymap.c:46]: This is unneeded and should be removed. 
```suggestion
```

File: keyboards/crkbd/keymaps/stecha/keymap.c [hunk 6]
```diff
@@ -0,0 +1,182 @@
...
+
+void matrix_init_user(void) {
+#ifdef RGBLIGHT_ENABLE
+    RGB_current_mode = rgblight_config.mode;
```
Comment (drashna) [inline keyboards/crkbd/keymaps/stecha/keymap.c:50]:
```suggestion
    RGB_current_mode = rgblight_get_mode();
```

File: keyboards/crkbd/keymaps/stecha/keymap.c [hunk 7]
```diff
@@ -0,0 +1,182 @@
...
+#ifdef SSD1306OLED
+        set_keylog(keycode, record);
+#endif
+        // set_timelog();
+    }
+
+    switch (keycode) {
+        case QWERTY:
+            if (record->event.pressed) {
+                persistent_default_layer_set(1UL << _QWERTY);
```
Comment (drashna) [inline keyboards/crkbd/keymaps/stecha/keymap.c:117]:
```suggestion
                set_single_persistent_default_layer(_QWERTY);
```

File: keyboards/crkbd/keymaps/stecha/keymap.c [hunk 8]
```diff
@@ -0,0 +1,182 @@
...
+                    tap_code16(LALT(KC_TAB));
+                }
+            }
+            return false;
+        case RGB_MOD:
+#ifdef RGBLIGHT_ENABLE
+            if (record->event.pressed) {
+                rgblight_mode(RGB_current_mode);
+                rgblight_step();
+                RGB_current_mode = rgblight_config.mode;
```
Comment (drashna) [inline keyboards/crkbd/keymaps/stecha/keymap.c:172]:
```suggestion
                RGB_current_mode = rgblight_get_mode();
```

File: keyboards/xd75/keymaps/stecha/keymap.c [hunk 9]
```diff
@@ -0,0 +1,122 @@
...
+            return false;
+    }
+    return true;
+}
+
+void matrix_init_user(void) {}
+
+void matrix_scan_user(void) {}
+
+void led_set_user(uint8_t usb_led) {}
```
Comment (drashna) [inline keyboards/xd75/keymaps/stecha/keymap.c:122]:
```suggestion
```

**Review by drashna — CHANGES_REQUESTED** [review]
You have a lot of shared code, in your keymaps.  You may want to consider using the userspace feature to share that code. 

https://docs.qmk.fm/#/feature_userspace

File: keyboards/boardwalk/keymaps/stecha/keymap.c (hunk 1 above)
Comment (SteKarab) [reply keyboards/boardwalk/keymaps/stecha/keymap.c:21]: Thanks for all the suggestions, drashna! Will make the required changes and look into userspaces as well.

**Pull request closed without merging**